core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/DataFeedServiceImpl.java

```diff
@@ -189,14 +188,27 @@ public void deleteDataFeed() throws DataFeedException {
             removePreviousDataFeed(serviceResourceResolver);
             removeCsvReport(serviceResourceResolver);
             removePendingNode(serviceResourceResolver);
-            gridResourcesCache.clearCache();
+            clearStaticDataFeed();
             serviceResourceResolver.commit();
         } catch (PersistenceException e) {
             LOG.error("Failed to delete data feed", e);
             throw new DataFeedException("Exception Deleting Data Feed");
         }
     }
 
+    private List<GridResource> getGridResourcesList() {
+        return new ArrayList<>(GRID_RESOURCE_COPY_ON_WRITE_ARRAY_LIST);
```

Comment:
What do you need to create  a new copy of the list every time - ?
It would only be justified if one of the consumer methods did modify the list (while all the rest counsumer methods would need it in the very original state). But as far as I can see, all of consumer methods just read it, or else completely replace it, but not modify